pkg/providers/tinkerbell/validate.go

```diff
@@ -27,22 +27,40 @@ func validateOsFamily(spec *ClusterSpec) error {
 		if spec.MachineConfigs[groupRef.Name].OSFamily() != controlPlaneOsFamily {
 			return fmt.Errorf("worker node group osFamily cannot be different from control plane osFamily")
 		}
-		if group.UpgradeRolloutStrategy != nil {
-			if spec.MachineConfigs[groupRef.Name].OSFamily() != v1alpha1.DefaultOSFamily && group.UpgradeRolloutStrategy.Type == "InPlace" {
-				return fmt.Errorf("InPlace upgrades are only supported on the Ubuntu OS family")
-			}
-		}
 	}
 
 	if controlPlaneOsFamily != v1alpha1.Bottlerocket && spec.DatacenterConfig.Spec.OSImageURL == "" && spec.ControlPlaneMachineConfig().Spec.OSImageURL == "" {
 		return fmt.Errorf("please use bottlerocket as osFamily for auto-importing or provide a valid osImageURL")
 	}
 
+	return nil
+}
+
+func validateUpgradeRolloutStrategy(spec *ClusterSpec) error {
+	var cpUpgradeRolloutStrategyType v1alpha1.UpgradeRolloutStrategyType
+
 	if spec.ControlPlaneConfiguration().UpgradeRolloutStrategy != nil {
-		if controlPlaneOsFamily != v1alpha1.DefaultOSFamily && spec.ControlPlaneConfiguration().UpgradeRolloutStrategy.Type == "InPlace" {
+		cpUpgradeRolloutStrategyType = spec.ControlPlaneConfiguration().UpgradeRolloutStrategy.Type
+		controlPlaneRef := spec.ControlPlaneConfiguration().MachineGroupRef
+		controlPlaneOsFamily := spec.MachineConfigs[controlPlaneRef.Name].OSFamily()
+
+		if controlPlaneOsFamily != v1alpha1.Ubuntu && cpUpgradeRolloutStrategyType == v1alpha1.InPlaceStrategyType {
 			return fmt.Errorf("InPlace upgrades are only supported on the Ubuntu OS family")
 		}
 	}
+
+	for _, group := range spec.Cluster.Spec.WorkerNodeGroupConfigurations {
+		groupRef := group.MachineGroupRef
+
+		if group.UpgradeRolloutStrategy != nil {
```

Comment:
Can the control plane strategy be nil while this is not? Or vice versa? Should we have another check for that or you think that is not needed?

Reply:
I think that is technically allowed because it defaults to `RollingUpdate` in CAPI if it isn't specified. But let's say they want to customize the maxSurge only on a worker node group, we don't have anything stopping that today.

Maybe here, I can set it to "RollingUpdate" at the beginning for this method since that's the default:
`cpUpgradeRolloutStrategyType := RollingUpdateStrategyType`
 
then if it is specified, we will set it to that instead. But if it isn't specified, this will still allow the worker nodes to pass the validation & potentially add more customization if desired. Thoughts?